tests/components/nws/test_nws.py

```diff
@@ -0,0 +1,337 @@
+import unittest
+from unittest.mock import patch
+
+import aiohttp
+
+from homeassistant.components import weather
+from homeassistant.components.nws.weather import ATTR_FORECAST_PRECIP_PROB
+from homeassistant.components.weather import (
+    ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_PRESSURE,
+    ATTR_WEATHER_TEMPERATURE,
+    ATTR_WEATHER_VISIBILITY,
+    ATTR_WEATHER_WIND_BEARING,
+    ATTR_WEATHER_WIND_SPEED,
+)
+from homeassistant.components.weather import (
+    ATTR_FORECAST,
+    ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+)
+
+from homeassistant.const import (
+    LENGTH_KILOMETERS,
+    LENGTH_METERS,
+    LENGTH_MILES,
+    PRECISION_WHOLE,
+    PRESSURE_INHG,
+    PRESSURE_PA,
+    PRESSURE_HPA,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.helpers.temperature import display_temp
+from homeassistant.util.pressure import convert as convert_pressure
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.unit_system import IMPERIAL_SYSTEM, METRIC_SYSTEM
+from homeassistant.util.temperature import convert as convert_temperature
+from homeassistant.setup import setup_component
+
+from tests.common import get_test_home_assistant, MockDependency
+
+OBS = {
+    "temperature": 7,
+    "relativeHumidity": 10,
+    "windDirection": 180,
+    "visibility": 10000,
+    "windSpeed": 10,
+    "seaLevelPressure": 30000,
+    "iconTime": "day",
+    "iconWeather": (("Fair/clear", None),),
+}
+
+FORE = [
+    {
+        "temperature": 41,
+        "windBearing": 180,
+        "windSpeedAvg": 9,
+        "iconTime": "day",
+        "startTime": "2018-12-21T15:00:00-05:00",
+        "iconWeather": (("Fair/Clear", None), ("Thunderstorm (high cloud cover)", 40)),
+    }
+]
+
+STN = "STNA"
+
+
+class MockNws:
+    """Mock Station from pynws."""
+
+    data_obs = None
+    data_fore = None
+
+    error_obs = False
+    error_fore = False
+    error_stn = False
+
+    def __init__(self, lat, lon, userid, mode, session):
+        """Init mock nws."""
+        self.station = None
+        self.stations = None
+
+    async def update_observation(self):
+        """Mock observation update."""
+        if self.error_obs:
+            raise aiohttp.ClientError
+        return
+
+    async def update_forecast(self):
+        """Mock forecast update."""
+        if self.error_fore:
+            raise aiohttp.ClientError
+        return
+
+    @property
+    def observation(self):
+        """Mock Observation."""
+        return self.data_obs
+
+    @property
+    def forecast(self):
+        """Mock Forecast."""
+        return self.data_fore
+
+    async def set_station(self, station=None):
+        """Mock stations."""
+        if self.error_stn:
+            raise aiohttp.ClientError
+        self.stations = [STN]
+        self.station = station or STN
+        return
+
+
+class TestNWS(unittest.TestCase):
```

Comment:
All new tests must be standalone pytest test functions. Ie not unittest class based test methods.

Reply:
I had based this off of one of the existing weather platforms, but this is useful feedback.  If I cannot reconfigure the tests in a timely manner, I will remove them and plan on adding them in a separate PR.